File: 프로그래머스/lv2/42842. 카펫/카펫.js
function solution(brown, yellow) {
    var answer = [];
    let sum = brown + yellow;
    
    for(let column = 3; column <= brown; column++){
        if(sum % column === 0){
            let row = sum / column;
            if((column-2) * (row-2) === yellow){
                return [row, column];
            }
        }
    }
    return answer;
}

File: 프로그래머스/lv2/42842. 카펫/카펫.py
def solution(brown, yellow):
    total = brown +yellow
    for column in range(3, int(total/3)+1):
        if total% column == 0:
            row = int(total/column)
            if row <= column and (column -2)*(row-2) == yellow:
                return [column, row]